cloudinit/importer.py

```diff
@@ -8,16 +8,34 @@
 #
 # This file is part of cloud-init. See LICENSE file for license information.
 
-import sys
+import importlib
+from types import ModuleType
+from typing import Optional, Sequence
 
 
-def import_module(module_name):
-    __import__(module_name)
-    return sys.modules[module_name]
+def import_module(module_name: str) -> ModuleType:
+    return importlib.import_module(module_name)
 
 
-def find_module(base_name: str, search_paths, required_attrs=None) -> tuple:
-    """Finds and imports specified modules"""
+def _count_attrs(
+    module_name: str, attrs: Optional[Sequence[str]] = None
+) -> int:
+    found_attrs = 0
+    if not attrs:
+        return found_attrs
+    mod = importlib.import_module(module_name)
+    for attr in attrs:
+        if hasattr(mod, attr):
+            found_attrs += 1
```

Comment:
Is the number of found attributes really the important thing to test in `find_module()`? In other words, if the module has every required attribute, why should it not be imported?

If the module has every required attribute, but has additional attributes that are not included in the required_attrs argument, won't this counting test cause import to fail? I agree that this looks functionally similar to what we're already doing, I wonder whether it is unnecessarily brittle. If I add an attribute to a module and now the runtime importer fails, I'd be confused - and having to dig into the magical import machinery to figure out what's going on is less than ideal for anyone writing a module.